File: Backend_Project/Project/AuctionAPI/Models/Item.cs
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;

namespace AuctionAPI.Models
{
    public class Item
    {
        [Key]
        public Guid Id { get; set; }

        public string Title { get; set; } = string.Empty;
        public string? Status { get; set; }
        public string? Category { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public Guid SellerID { get; set; }
        public Guid? BidderID { get; set; }
        public bool IsDeleted { get; set; } = false;
        public Collection<Bid>? Bids { get; set; }
        public Seller? Seller { get; set; }
        public Bidder? Bidder { get; set; }
        public ItemDetails? ItemDetails { get; set; }
    }
}

File: Frontend_Project/AuctionAPI/src/app/models/item.ts
interface Item {
  itemID: string;
  title: string;
  description: string;
  status: string;
  category: string;
  startingPrice: number;
  startDate: string;
  endDate: string;
  currentBid?: number;
  currentBidderName?: string;
  sellerId?: string;
  sellerName: string;
  imageUrl?: string;
  boughtBy?: string;
}